pkg/test/case.go

```diff
@@ -370,7 +371,18 @@ func (t *Case) Run(test *testing.T, ts *report.Testsuite) {
 		tc.Assertions += len(testStep.Asserts)
 		tc.Assertions += len(testStep.Errors)
 
-		errs := testStep.Run(test, ns.Name)
+		errs := []error{}
+
+		if testStep.KubeconfigLoading == v1beta1.KubeconfigLoadingLazy {
+			cl, err = newClient(testStep.Kubeconfig)(false)
+			if err != nil {
+				errs = append(errs, fmt.Errorf("failed to lazy-load kubeconfig %q: %w", testStep.Kubeconfig, err))
+			} else {
+				clients[testStep.Kubeconfig] = cl
+			}
+		}
+
+		errs = append(errs, testStep.Run(test, ns.Name)...)
```

Comment:
I noticed in the output that we're getting the error twice now:

```
    case.go:392: failed to lazy-load kubeconfig "/Users/kumarmallikarjuna/Workspace/kuttl/lazy_test/lazy/loading/test.kubeconfig": stat /Users/kumarmallikarjuna/Workspace/kuttl/lazy_test/lazy/loading/test.kubeconfig: no such file or directory
    case.go:392: stat /Users/kumarmallikarjuna/Workspace/kuttl/lazy_test/lazy/loading/test.kubeconfig: no such file or directory
```

I guess this is because we `Run` the step on this line regardless of whether the lazy loading succeeded or not 😟 

However looking deeper into this I realized a few more problems that I didn't see earlier, sorry:
- the conditional creation of a client on line 377 is mostly redundant since we already prepare one on line 364,
- more importantly, we're failing to create a namespace for lazy-loaded clients, I think we should replace the block  377-382 with one that creates a namespace if it hasn't been done eagerly (not sure how to do this elegantly and robustly, but I'm sure you can come up with something 😉 )

Reply:
> I noticed in the output that we're getting the error twice now:
> 
> ```
>     case.go:392: failed to lazy-load kubeconfig "/Users/kumarmallikarjuna/Workspace/kuttl/lazy_test/lazy/loading/test.kubeconfig": stat /Users/kumarmallikarjuna/Workspace/kuttl/lazy_test/lazy/loading/test.kubeconfig: no such file or directory
>     case.go:392: stat /Users/kumarmallikarjuna/Workspace/kuttl/lazy_test/lazy/loading/test.kubeconfig: no such file or directory
> ```
> 
> I guess this is because we `Run` the step on this line regardless of whether the lazy loading succeeded or not 😟

Fixed here: https://github.com/kudobuilder/kuttl/pull/540/commits/75eb6a9b3fc25634b10b2b7c9d5022e2d9b16a1d. I've skipped the test setup from running in-case the kubeconfig setup fails.

> However looking deeper into this I realized a few more problems that I didn't see earlier, sorry:
> 
> * the conditional creation of a client on line 377 is mostly redundant since we already prepare one on line 364,
> * more importantly, we're failing to create a namespace for lazy-loaded clients, I think we should replace the block  377-382 with one that creates a namespace if it hasn't been done eagerly (not sure how to do this elegantly and robustly, but I'm sure you can come up with something 😉 )

Totally missed that. Since, the namespace is created for each eager client, I've replicated that and placed the logic right after the lazy-client is created:
https://github.com/kudobuilder/kuttl/pull/540/commits/82fe683e5317454767fced3e51f517d1232164cd.

Reply:
Possible edge case here: the Kubeconfigs are common b/w steps or the same as the global one. In that case, the namespace would attempt to be re-created and its failure would keep the test from evaluating. I'll add an additional check.